receiver/jaegerlegacyreceiver/factory.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jaegerlegacyreceiver
+
+// This file implements factory for Jaeger receiver.
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/open-telemetry/opentelemetry-collector/config/configerror"
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/receiver"
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "jaegerlegacy"
```

Comment:
nit: we have been using underscores on this type of config entry
```suggestion
	typeStr = "jaeger_legacy"
```